src/com/seafile/seadroid2/ui/adapter/SeafItemAdapter.java

```diff
@@ -284,41 +290,17 @@ private void setFileView(SeafDirent dirent, Viewholder viewHolder, int position)
 
             viewHolder.downloadStatusIcon.setImageResource(downloadStatusIcon);
             viewHolder.subtitle.setText(dirent.getSubtitle());
+        }
+
+        if (Utils.isViewableImage(file.getName())) {
```

Comment:
I think we can use a flag to mark if thumbnails were downloaded already . If yes, don\`t bother to poke the server again.